pkg/apis/extensions/register.go

```diff
@@ -63,8 +64,8 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 		&ReplicaSetList{},
 		&PodSecurityPolicy{},
 		&PodSecurityPolicyList{},
-		&NetworkPolicy{},
-		&NetworkPolicyList{},
+		&networking.NetworkPolicy{},
```

Comment:
Right, to the outside nothing changes. We only merge the internal types. We do exactly the same trick with autoscaling and apps (unfortunately still conversely, i.e. internal types are in extensions; we should change that).

About your question: yes we need the internal types being registered under extensions if the external types should work for that group. We never switch to another internal group during conversion. Technically, the types coming out of such a conversion will be the internal types from the networking group.